src/UnityExtension/Assets/Editor/GitHub.Unity/UI/GitPathView.cs

```diff
@@ -0,0 +1,286 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEditor;
+using UnityEngine;
+
+namespace GitHub.Unity
+{
+    [Serializable]
+    class GitPathView : Subview
+    {
+        private const string GitInstallTitle = "Git installation";
+        private const string PathToGit = "Path to Git";
+        private const string GitPathSaveButton = "Save Path";
+        private const string GitInstallFindButton = "Find install";
+        private const string BrowseButton = "...";
+        private const string GitInstallBrowseTitle = "Select git binary";
+        private const string ErrorInvalidPathMessage = "Invalid Path.";
+        private const string ErrorGettingSoftwareVersionMessage = "Error getting software versions.";
+        private const string ErrorMinimumGitVersionMessageFormat = "Git version {0} found. Git version {1} is required.";
+        private const string ErrorMinimumGitLfsVersionMessageFormat = "Git LFS version {0} found. Git LFS version {1} is required.";
+
+        [SerializeField] private string gitExec;
+        [SerializeField] private string gitExecParent;
+        [SerializeField] private string gitExecExtension;
+        [SerializeField] private string newGitExec;
+        [SerializeField] private bool isValueChanged;
+        [SerializeField] private bool isValueChangedAndFileExists;
+        [SerializeField] private string gitFileErrorMessage;
+        [SerializeField] private string gitVersionErrorMessage;
+
+        [NonSerialized] private bool isBusy;
+        [NonSerialized] private bool gitExecHasChanged;
+
+        public override void OnEnable()
+        {
+            base.OnEnable();
+
+            gitExecHasChanged = true;
+        }
+
+        public override void OnDataUpdate()
+        {
+            base.OnDataUpdate();
+            MaybeUpdateData();
+        }
+
+        public override bool IsBusy
+        {
+            get { return isBusy; }
+        }
+
+        public override void OnGUI()
+        {
+            // Install path
+            GUILayout.Label(GitInstallTitle, EditorStyles.boldLabel);
+
+            EditorGUI.BeginDisabledGroup(IsBusy || Parent.IsBusy);
+            {
+                // Install path field
+                GUILayout.BeginHorizontal();
+                {
+                    EditorGUI.BeginChangeCheck();
+                    {
+                        newGitExec = EditorGUILayout.TextField(PathToGit, newGitExec);
+                    }
+
+                    if (EditorGUI.EndChangeCheck())
+                    {
+                        CheckEnteredGitPath();
+                    }
+
+                    if (GUILayout.Button(BrowseButton, EditorStyles.miniButton, GUILayout.Width(25)))
```

Comment:
Width should be defined in a style (either here or in the Styles class) and not hardcoded here

Reply:
👍